Editor/XRUtils/XRLoaderHelper.cs

```diff
@@ -60,102 +88,101 @@ public static void LoadOculus()
         /// </summary>
         public static void LoadWindowsMR()
         {
-#if CREATOR_XR_MANAGEMENT
-            AddScriptingDefineSymbol(WindowsMRDefineSymbol);
+            if (GetCurrentXRConfiguration().Any(loader => loader == XRConfiguration.WindowsMR))
+            {
+                Debug.LogWarning("Windows MR is already loaded.");
+                return;
+            }
+
+#if UNITY_XR_MANAGEMENT
+            DisplayDialog("Windows MR");
+            PackageOperationsManager.LoadPackage(WindowsXRPackage);
 #else
+            DisplayDialog("XR Plug-in Management");
             EditorPrefs.SetInt(nameof(XRSDK), (int)XRSDK.WindowsMR);
-            AddScriptingDefineSymbol(XRManagementDefineSymbol);
+            PackageOperationsManager.LoadPackage(XRManagementPackage);
 #endif
         }
 
-#if UNITY_2019_1_OR_NEWER && !UNITY_2020_1_OR_NEWER
-        private async static void LoadOpenVRLegacy ()
-        {
-#pragma warning disable CS0618
-            PlayerSettings.virtualRealitySupported = true;
-            UnityEngine.XR.XRSettings.LoadDeviceByName("OpenVR");
-            await System.Threading.Tasks.Task.Delay(1000);
-            UnityEngine.XR.XRSettings.enabled = true;
-#pragma warning restore CS0618
-        }
-#endif
-
-        private static void AddScriptingDefineSymbol(string scriptingDefineSymbol)
+        /// <summary>
+        /// Returns a list with all XR SDKs enabled in this project.
+        /// </summary>
+        public static IEnumerable<XRConfiguration> GetCurrentXRConfiguration()
         {
-            BuildTarget buildTarget = BuildTarget.StandaloneWindows;
-            BuildTargetGroup buildTargetGroup = BuildPipeline.GetBuildTargetGroup(buildTarget);
-            List<string> symbols = PlayerSettings.GetScriptingDefineSymbolsForGroup(buildTargetGroup).Split(';').ToList();
+            List<XRConfiguration> enabledSDKs = new List<XRConfiguration>();
 
-            if (symbols.Contains(scriptingDefineSymbol) == false)
+            if (EditorReflectionUtils.AssemblyExists("Unity.XR.Management"))
             {
-                symbols.Add(scriptingDefineSymbol);
-                PlayerSettings.SetScriptingDefineSymbolsForGroup(buildTargetGroup, string.Join(";", symbols.ToArray()));
-            }
-        }
+#if UNITY_XR_MANAGEMENT
```

Comment:
The combination of ifs and compiler directives makes it hard to read.

Reply:
Yes I know 😞
I am usually against using them unless necessary, let me try to optimize it maybe in different classes so it gets easier to read.

In general, the work flow is super confused and has to be done this way due to different factors, but mainly not all assemblies are present at the start and there are things that have to be started as soon as an assembly is loaded.

For example, in order to load Oculus, it might be required to load first the XR Management package, then Oculus, then enable Oculus in the XR management system 😱